src/modules/vendor/components/courses/CourseCreationModal.vue

```diff
@@ -88,12 +88,12 @@ export default {
     submit () {
       this.$emit('submit');
     },
+    updateType (event) {
+      delete this.newCourse.company;
```

Comment:
du coup tu peux enlever cette ligne puisque c'est geré par le `this.update(undefined, 'company');`
et en plus si tu le supprimes ici tu modifies la props dans l'enfant donc ca va a l'encontre de la regle qu'on est en train de corriger

Reply:
etrange quand je fais mes deux appels seul le deuxieme réussi le premier semble ignoré quelquesoit l'ordre
```
      this.update(event, 'type');
      this.update(undefined, 'company');
```